### DISCUSSION TITLE
Multi-cursor editing support in search result buffers

### DISCUSSION THREAD

**Discussion by bobosola** [zed-industries/zed#6533] [Feature Requests]
How to activate multi-line cursors was answered in https://github.com/zed-industries/community/discussions/1075 - I wouldn't have known the feature exists had I not read that post, as the feature seems to be undocumented and not included in the command palette?

A (possibly) related question is: how to do we activate the very interesting multi-buffer edit cursors as demoed by Nathan Sobo at [https://youtu.be/wXT73bBr83s?t=1142](https://youtu.be/wXT73bBr83s?t=1142) where he does a project-wide search then opens a live editing cursor in each of the search result buffer excerpts?  Is this feature still in the current version, and if so, how do we activate it please?

**Comment by JosephTLyons** [comment]
Hey @bobosola, I think Nathan ran a `editor: select next` command a handful of times (by default, this is bound to `cmd-d`).  Each time you run that command, the editor will highlight the next occurrence that matches whatever you have currently highlighted.  Then, he probably ran a command to jump all cursors to the end of the line (`cmd-right`), where he then typed in a comment.

https://user-images.githubusercontent.com/19867440/231612063-dfbf1d19-7499-46fb-87b8-61d1c1e2268d.mov

Just to mention it, these commands can be ran in any editor, they aren't restricted to only multi-buffers.

We also do have a [request to add a command that selects all of the matching text with one command](https://github.com/zed-industries/community/issues/75), but this hasn't landed yet.

**Reply by bobosola** [reply]
Explains it perfectly, thanks for the fast reply, much appreciated.
